File: 0200-number-of-islands/0200-number-of-islands.cs
public class Solution {
    public int NumIslands(char[][] grid) {
        int m = grid.Length, n = grid[0].Length, count = 0;
        for(var i=0; i<m; i++)
            for(var j=0; j<n; j++)
                if(grid[i][j] == '1'){
                    count++;
                    TraverseIsland(grid, i, j);
                }
        
        return count;
    }

    // DFS
    private void TraverseIsland(char[][] grid, int x, int y) {
        if(x == -1 || y == -1 || x == grid.Length || y == grid[0].Length || grid[x][y] == '0') return;
        grid[x][y] = '0';
        TraverseIsland(grid, x-1, y);
        TraverseIsland(grid, x+1, y);
        TraverseIsland(grid, x, y-1);
        TraverseIsland(grid, x, y+1);
    }
}

File: 0200-number-of-islands/0200-number-of-islands.py
class Solution:
    def numIslands(self, grid: List[List[str]]) -> int:
        count = 0
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == '1':
                    count += 1
                    self.traverseIsland(grid, i, j)
        return count

    # DFS
    def traverseIsland(self, grid: List[List[str]], x: int, y: int):
        if(x == -1 or y == -1 or x == len(grid) or y == len(grid[0]) or grid[x][y] == '0'): return
        grid[x][y] = '0'
        self.traverseIsland(grid, x-1, y)
        self.traverseIsland(grid, x+1, y)
        self.traverseIsland(grid, x, y-1)
        self.traverseIsland(grid, x, y+1)